wpsimplepay/wp-simple-pay-lite
# Cherry Pick In-Plugin Upgrade Promos

Pulling from Pro:
https://github.com/wpsimplepay/wp-simple-pay-pro/issues/1697
https://github.com/wpsimplepay/wp-simple-pay-pro/pull/1698

The only changes this brings in is the in-plugin top of page upgrade promo. 

To test:

_Observe existing permanent notice dismissal continues to function as expected_

1. Disconnect from Stripe Connect to display the "Connect with Stripe" notice. (Clear `simpay_dismiss_stripe_connect` from `wp_options` if previously dismissed).
2. Click the `(x)` button to dismiss the notice.
3. Refresh the page.

- Observe the notice remains dismissed.

---

_Observe existing upgrade promos continue to function as expected_

1. Edit a Payment Form

- Observe the "Want to customize your payment forms even more" promo is displayed under each tab.

Note: These UTM params are different than what is used in the top of page promo but these will be dealt with in https://github.com/wpsimplepay/wp-simple-pay-pro/issues/1713

There is also a slightly larger extra margin around the promo but it is fine to leave as-is to minimize the code change footprint.

---

_Observe new top of page upgrade promo_

1. Visit a WP Simple Pay page, such as the list of Payment Forms.

- Observe the "Your are using WP Simple Pay Lite..." promo slides in
- Observe the "Purchase a license" hyperlink points to `https://wpsimplepay.com/lite-vs-pro/?utm_source=settings&utm_medium=upgrade-from-lite&utm_campaign=admin&utm_content=link-1`

2. Refresh the page.

- Observe the "Your are using WP Simple Pay Lite..." promo appears instantly.

3. Dismiss the notice.
4. Refresh the page.

- Observe no promo is shown.
